### ISSUE TITLE
ArgumentError: new key RB_USER_INSTALL

### ISSUE THREAD

**Issue by vtamara** [bundler/bundler#6013]
--- ERROR REPORT TEMPLATE -------------------------------------------------------
# Error Report

## Questions

Please fill out answers to these questions, it'll help us figure out
why things are going wrong.

- **What did you do?**

  I ran the command `/usr/local/bin/bundle install`

- **What did you expect to happen?**

 Install gems

- **What happened instead?**

Error

- **Have you tried any solutions posted on similar issues in our issue tracker, stack overflow, or google?**

   I was testing bundler 1.16.0.pre.2, will go back to bundler 1.15

- **Have you read our issues document, https://github.com/bundler/bundler/blob/master/doc/contributing/ISSUES.md?**

  No

## Backtrace

```
ArgumentError: new key RB_USER_INSTALL
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/shared_helpers.rb:279:in `set_env'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/cli/install.rb:17:in `run'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/cli.rb:224:in `block in install'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/settings.rb:135:in `temporary'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/cli.rb:223:in `install'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/vendor/thor/lib/thor/invocation.rb:126:in `invoke_command'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/vendor/thor/lib/thor.rb:387:in `dispatch'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/cli.rb:27:in `dispatch'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/vendor/thor/lib/thor/base.rb:466:in `start'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/cli.rb:18:in `start'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/exe/bundle:30:in `block in <top (required)>'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/lib/bundler/friendly_errors.rb:122:in `with_friendly_errors'
  /usr/local/lib/ruby/gems/2.4/gems/bundler-1.16.0.pre.2/exe/bundle:22:in `<top (required)>'
  /usr/local/bin/bundle:22:in `load'
  /usr/local/bin/bundle:22:in `<main>'
```

## Environment

```
Bundler   1.16.0.pre.2
RubyGems  2.6.11
Ruby      2.4.1p111 (2017-03-22 revision 58053) [x86_64-openbsd]
Git       2.9.0
Platform  x86_64-openbsd
OpenSSL   LibreSSL 2.4.5
```

## Bundler Build Metadata

```
Built At          2017-09-07
Git SHA           c1db119a5
Released Version  true
```

## Bundler settings

```
build.nokogiri
  Set for the current user (/home/vtamara/.bundle/config): "--use-system-libraries --with-xml2-include=/usr/local/include/libxml2/"
path
  Set for your local app (/home/vtamara/comp/rails/heb412_gen/.bundle/config): "/var/www/bundler"
disable_share_gems
  Set for your local app (/home/vtamara/comp/rails/heb412_gen/.bundle/config): "true"
disable_shared_gems
  Set for your local app (/home/vtamara/comp/rails/heb412_gen/.bundle/config): true
```

### ../../Gemfile

```ruby
source 'https://rubygems.org'


# Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
gem 'rails', '~> 5.1.0'
# Use sqlite3 as the database for Active Record
gem 'pg'
# Use Puma as the app server
gem 'puma'

gem 'colorize'
# Use SCSS for stylesheets
gem 'sass'
gem 'sass-rails'

gem 'chosen-rails'

# Use Uglifier as compressor for JavaScript assets
gem 'uglifier', '>= 1.3.0'
# Use CoffeeScript for .coffee assets and views
gem 'coffee-rails', '~> 4.2'
# See https://github.com/rails/execjs#readme for more supported runtimes
# gem 'therubyracer', platforms: :ruby

# Use jquery as the JavaScript library
gem 'jquery-rails'
gem 'jquery-ui-rails'
# Turbolinks makes navigating your web application faster. Read more: https://github.com/turbolinks/turbolinks


# Build JSON APIs with ease. Read more: https://github.com/rails/jbuilder
gem 'jbuilder', '~> 2.5'
# Use Redis adapter to run Action Cable in production
# gem 'redis', '~> 3.0'
# Use ActiveModel has_secure_password
# gem 'bcrypt', '~> 3.1.7'

# Use Capistrano for deployment
# gem 'capistrano-rails', group: :development


# Ambiente de CSS
gem "twitter-bootstrap-rails"
gem "bootstrap-datepicker-rails"
gem "font-awesome-rails"

# Formularios simples
gem "simple_form"

# Formularios anidados (algunos con ajax)
gem "cocoon", git: "https://github.com/vtamara/cocoon.git"

# Autenticación y roles
gem "devise"
gem "devise-i18n"
gem "cancancan"
gem "bcrypt"
# Listados en páginas
gem "will_paginate"

# ICU con CLDR
gem 'twitter_cldr'

# Maneja adjuntos
gem "paperclip"#, "~> 4.1"

gem 'libxml-ruby'
gem 'rspreadsheet'


# Zonas horarias
gem "tzinfo"

# Motor de sistemas de información estilo Pasos de Jesús
gem 'sip', git: "https://github.com/pasosdeJesus/sip.git", branch: :us_modelo
#gem 'sip', path: '../sip'

group :development, :test do
  # Call 'byebug' anywhere in the code to stop execution and get a debugger console
  #gem 'byebug', platform: :mri
end

group :test do

  gem 'simplecov'
  # Envia resultados de pruebas desde travis a codeclimate
end

group :development do
  # Access an IRB console on exception pages or by using <%= console %> anywhere in the code.
  gem 'web-console'
  # Spring speeds up development by keeping your application running in the background. Read more: https://github.com/rails/spring
  gem 'spring'
end
```

### ../../Gemfile.lock

```
GIT
  remote: https://github.com/pasosdeJesus/sip.git
  revision: 8b743136a5907898bd0f05ead47eeea8236d08c6
  branch: us_modelo
  specs:
    sip (2.0.b4)
      bootstrap-datepicker-rails
      cancancan
      chosen-rails
      colorize
      devise
      devise-i18n
      jquery-rails
      jquery-ui-rails
      minitest
      paperclip
      rails
      rails-i18n
      simple_form
      twitter-bootstrap-rails
      twitter_cldr
      will_paginate

GIT
  remote: https://github.com/vtamara/cocoon.git
  revision: 20f8f3bab2fee21309c57b586f5fab9921ac0894
  specs:
    cocoon (1.2.6)

GEM
  remote: https://rubygems.org/
  specs:
    actioncable (5.1.3)
      actionpack (= 5.1.3)
      nio4r (~> 2.0)
      websocket-driver (~> 0.6.1)
    actionmailer (5.1.3)
      actionpack (= 5.1.3)
      actionview (= 5.1.3)
      activejob (= 5.1.3)
      mail (~> 2.5, >= 2.5.4)
      rails-dom-testing (~> 2.0)
    actionpack (5.1.3)
      actionview (= 5.1.3)
      activesupport (= 5.1.3)
      rack (~> 2.0)
      rack-test (~> 0.6.3)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.2)
    actionview (5.1.3)
      activesupport (= 5.1.3)
      builder (~> 3.1)
      erubi (~> 1.4)
      rails-dom-testing (~> 2.0)
      rails-html-sanitizer (~> 1.0, >= 1.0.3)
    activejob (5.1.3)
      activesupport (= 5.1.3)
      globalid (>= 0.3.6)
    activemodel (5.1.3)
      activesupport (= 5.1.3)
    activerecord (5.1.3)
      activemodel (= 5.1.3)
      activesupport (= 5.1.3)
      arel (~> 8.0)
    activesupport (5.1.3)
      concurrent-ruby (~> 1.0, >= 1.0.2)
      i18n (~> 0.7)
      minitest (~> 5.1)
      tzinfo (~> 1.1)
    andand (1.3.3)
    arel (8.0.0)
    bcrypt (3.1.11)
    bindex (0.5.0)
    bootstrap-datepicker-rails (1.7.1.1)
      railties (>= 3.0)
    builder (3.2.3)
    camertron-eprun (1.1.1)
    cancancan (2.0.0)
    chosen-rails (1.5.2)
      coffee-rails (>= 3.2)
      railties (>= 3.0)
      sass-rails (>= 3.2)
    cldr-plurals-runtime-rb (1.0.1)
    climate_control (0.2.0)
    cocaine (0.5.8)
      climate_control (>= 0.0.3, < 1.0)
    coffee-rails (4.2.2)
      coffee-script (>= 2.2.0)
      railties (>= 4.0.0)
    coffee-script (2.4.1)
      coffee-script-source
      execjs
    coffee-script-source (1.12.2)
    colorize (0.8.1)
    commonjs (0.2.7)
    concurrent-ruby (1.0.5)
    devise (4.3.0)
      bcrypt (~> 3.0)
      orm_adapter (~> 0.1)
      railties (>= 4.1.0, < 5.2)
      responders
      warden (~> 1.2.3)
    devise-i18n (1.2.0)
    docile (1.1.5)
    erubi (1.6.1)
    execjs (2.7.0)
    ffi (1.9.18)
    font-awesome-rails (4.7.0.2)
      railties (>= 3.2, < 5.2)
    globalid (0.4.0)
      activesupport (>= 4.2.0)
    i18n (0.8.6)
    jbuilder (2.7.0)
      activesupport (>= 4.2.0)
      multi_json (>= 1.2)
    jquery-rails (4.3.1)
      rails-dom-testing (>= 1, < 3)
      railties (>= 4.2.0)
      thor (>= 0.14, < 2.0)
    jquery-ui-rails (6.0.1)
      railties (>= 3.2.16)
    json (2.1.0)
    less (2.6.0)
      commonjs (~> 0.2.7)
    less-rails (2.8.0)
      actionpack (>= 4.0)
      less (~> 2.6.0)
      sprockets (> 2, < 4)
      tilt
    libxml-ruby (3.0.0)
    loofah (2.0.3)
      nokogiri (>= 1.5.9)
    mail (2.6.6)
      mime-types (>= 1.16, < 4)
    method_source (0.8.2)
    mime-types (3.1)
      mime-types-data (~> 3.2015)
    mime-types-data (3.2016.0521)
    mimemagic (0.3.2)
    mini_portile2 (2.2.0)
    minitest (5.10.3)
    multi_json (1.12.2)
    nio4r (2.1.0)
    nokogiri (1.8.0)
      mini_portile2 (~> 2.2.0)
    orm_adapter (0.5.0)
    paperclip (5.1.0)
      activemodel (>= 4.2.0)
      activesupport (>= 4.2.0)
      cocaine (~> 0.5.5)
      mime-types
      mimemagic (~> 0.3.0)
    pg (0.21.0)
    puma (3.10.0)
    rack (2.0.3)
    rack-test (0.6.3)
      rack (>= 1.0)
    rails (5.1.3)
      actioncable (= 5.1.3)
      actionmailer (= 5.1.3)
      actionpack (= 5.1.3)
      actionview (= 5.1.3)
      activejob (= 5.1.3)
      activemodel (= 5.1.3)
      activerecord (= 5.1.3)
      activesupport (= 5.1.3)
      bundler (>= 1.3.0)
      railties (= 5.1.3)
      sprockets-rails (>= 2.0.0)
    rails-dom-testing (2.0.3)
      activesupport (>= 4.2.0)
      nokogiri (>= 1.6)
    rails-html-sanitizer (1.0.3)
      loofah (~> 2.0)
    rails-i18n (5.0.4)
      i18n (~> 0.7)
      railties (~> 5.0)
    railties (5.1.3)
      actionpack (= 5.1.3)
      activesupport (= 5.1.3)
      method_source
      rake (>= 0.8.7)
      thor (>= 0.18.1, < 2.0)
    rake (12.0.0)
    rb-fsevent (0.10.2)
    rb-inotify (0.9.10)
      ffi (>= 0.5.0, < 2)
    responders (2.4.0)
      actionpack (>= 4.2.0, < 5.3)
     railties (>= 4.2.0, < 5.3)
    rspreadsheet (0.4.5)
      andand (~> 1.3)
      rubyzip (~> 1.1)
    rubyzip (1.2.1)
    sass (3.5.1)
      sass-listen (~> 4.0.0)
    sass-listen (4.0.0)
      rb-fsevent (~> 0.9, >= 0.9.4)
      rb-inotify (~> 0.9, >= 0.9.7)
    sass-rails (5.0.6)
      railties (>= 4.0.0, < 6)
      sass (~> 3.1)
      sprockets (>= 2.8, < 4.0)
      sprockets-rails (>= 2.0, < 4.0)
      tilt (>= 1.1, < 3)
    simple_form (3.5.0)
      actionpack (> 4, < 5.2)
      activemodel (> 4, < 5.2)
    simplecov (0.15.0)
      docile (~> 1.1.0)
      json (>= 1.8, < 3)
      simplecov-html (~> 0.10.0)
    simplecov-html (0.10.2)
    spring (2.0.2)
      activesupport (>= 4.2)
    sprockets (3.7.1)
      concurrent-ruby (~> 1.0)
      rack (> 1, < 3)
    sprockets-rails (3.2.1)
      actionpack (>= 4.0)
      activesupport (>= 4.0)
      sprockets (>= 3.0.0)
    thor (0.20.0)
    thread_safe (0.3.6)
    tilt (2.0.8)
    turbolinks (5.0.1)
      turbolinks-source (~> 5)
    turbolinks-source (5.0.3)
    twitter-bootstrap-rails (4.0.0)
      actionpack (~> 5.0, >= 5.0.1)
      execjs (~> 2.7)
      less-rails (~> 2.8, >= 2.8.0)
      railties (~> 5.0, >= 5.0.1)
    twitter_cldr (4.4.2)
      camertron-eprun
      cldr-plurals-runtime-rb (~> 1.0)
      tzinfo
    tzinfo (1.2.3)
      thread_safe (~> 0.1)
    uglifier (3.2.0)
      execjs (>= 0.3.0, < 3)
    warden (1.2.7)
      rack (>= 1.0)
    web-console (3.5.1)
      actionview (>= 5.0)
      activemodel (>= 5.0)
      bindex (>= 0.4.0)
      railties (>= 5.0)
    websocket-driver (0.6.5)
      websocket-extensions (>= 0.1.0)
    websocket-extensions (0.1.2)
    will_paginate (3.1.6)

PLATFORMS
  ruby

DEPENDENCIES
  bcrypt
  bootstrap-datepicker-rails
  cancancan
  chosen-rails
  cocoon!
  coffee-rails (~> 4.2)
  colorize
  devise
  devise-i18n
  font-awesome-rails
  jbuilder (~> 2.5)
  jquery-rails
  jquery-ui-rails
  libxml-ruby
  paperclip
  pg
  puma
  rails (~> 5.1.0)
  rspreadsheet
  sass
  sass-rails
  simple_form
  simplecov
  sip!
  spring
  turbolinks
  twitter-bootstrap-rails
  twitter_cldr
  tzinfo
  uglifier (>= 1.3.0)
  web-console
  will_paginate

BUNDLED WITH
   1.16.0.pre.2
```

--- TEMPLATE END ----------------------------------------------------------------

**Comment by olleolleolle** [comment]
@vtamara Does this FreeBSD-related issue (2010!) ring a bell?

https://github.com/wycats/bundler/issues/163

**Comment by segiddins** [comment]
@oleolleolle  I think it just needs to go into the environment preserver list

**Comment by segiddins** [comment]
Any interest in sending a PR for that?

**Comment by olleolleolle** [comment]
Fixed in #6015

**Comment by vtamara** [comment]
Today I tried the master branch of bundler and I confirm this error is not happening anymore. Thank you.
